shared/Microsoft.Extensions.StackTrace.Sources/StackFrame/StackTraceHelper.cs

```diff
@@ -119,5 +143,81 @@ internal static MethodDisplayInfo GetMethodDisplayString(MethodBase method)
 
             return methodDisplayInfo;
         }
+
+        private static bool ShowInStackTrace(MethodBase method)
+        {
+            Debug.Assert(method != null);
+
+            // Don't show any methods marked with the StackTraceHiddenAttribute
+            // https://github.com/dotnet/coreclr/pull/14652
+            foreach (var attibute in method.CustomAttributes)
+            {
+                // internal Attribute, match on name
+                if (attibute.AttributeType.Name == "StackTraceHiddenAttribute")
```

Comment:
Should we look up the full name? i.e. `System.Diagnostics.StackTraceHiddenAttribute`. Also (`string.Equals(StringComparison.Ordinal)`

Reply:
`operator ==(String a, String b)` is Ordinal?

`operator ==(String a, String b)` => `Equals(String a, String b)` => `EqualsHelper(a, b);`

Which is the same function `String.Equals(String a, String b, StringComparison.Ordinal)` ends up calling: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/String.Comparison.cs#L904-L952

Reply:
> Should we look up the full name? i.e. System.Diagnostics.StackTraceHiddenAttribute.

Can't add your own then :) Also much longer string to match on and if there was a second one its intent would likely be the same?

Will aim to get it expose via api review; then can refer to the type directly; and everyone can use it for this purpose (and revisit the matching as can be more efficient then)

Reply:
Huh, always thought == did InvariantCulture. Nevermind then